practice1/eurodiffusion/eurodiffusion.go

```diff
@@ -0,0 +1,144 @@
+package eurodiffusion
+
+import (
+	"math"
+
+	"github.com/akrava/software-engineering-methodology-practices/practice1/common"
+)
+
+func CalculateEuroDiffusionForTestCase(countries common.CountryList) common.TestCaseResults {
+	minXl, minYl, maxYh, maxXh := getMinXlMinYlMaxYhMaxMh(countries)
+	countRows := maxYh - minYl + 1
+	countCols := maxXh - minXl + 1
+	// initialize trimmed grid of cities
+	citiesGrid := make([][]*common.City, countRows)
+	for y := range citiesGrid {
+		citiesGrid[y] = make([]*common.City, countCols)
+	}
+	// put cities into grid
+	offsetX := minXl
+	offsetY := minYl
+	for _, country := range countries {
+		country.InitCities()
+		for _, city := range country.Cities {
+			citiesGrid[city.Y-offsetY][city.X-offsetX] = city
+		}
+	}
+	// start euro diffusion
+	countDays := 0
+	for {
```

Comment:
Would it be an infinite loop in case when there is no neighbour city?

Reply:
Created issue #16 from this comment. Thanks!

Reply:
Issue #16 is closed. I'll mark this conversation as resolved. Feel free to reopen if it's needed